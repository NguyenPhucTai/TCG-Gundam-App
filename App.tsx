import React from 'react';
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';

export default function App() {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>TCG Gundam Score App</Text>
      <Text style={styles.subtitle}>Board Game Score Tracker</Text>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 10,
    color: '#333',
  },
  subtitle: {
    fontSize: 16,
    color: '#666',
  },
});
