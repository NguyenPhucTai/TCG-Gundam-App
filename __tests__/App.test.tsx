import React from 'react';
import { render, screen } from '@testing-library/react-native';
import App from '../App';

describe('App Component', () => {
  it('renders correctly', () => {
    const tree = render(<App />);
    expect(tree).toBeTruthy();
  });

  it('should display the main text', () => {
    render(<App />);
    
    // Kiểm tra text có tồn tại
    const mainText = screen.getByText('Open up App.tsx to start working on your app!');
    expect(mainText).toBeTruthy();
  });

  it('should render the main container', () => {
    const { getByText } = render(<App />);
    
    // Kiểm tra container chứa text
    const container = getByText('Open up App.tsx to start working on your app!').parent;
    expect(container).toBeTruthy();
  });

  it('should match snapshot', () => {
    const tree = render(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
