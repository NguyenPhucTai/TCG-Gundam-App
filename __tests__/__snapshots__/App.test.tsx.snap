// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App Component should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
  <StatusBar
    style="auto"
  />
</View>
`;
